kmip/pie/api.py

```diff
@@ -0,0 +1,95 @@
+# Copyright (c) 2015 The Johns Hopkins University/Applied Physics Laboratory
+# All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License. You may obtain
+# a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations
+# under the License.
+
+import abc
+import six
+
+
+@six.add_metaclass(abc.ABCMeta)
+class KmipClientAPI:
```

Comment:
Ok, that makes sense. Here are alternative names for the actual `KmipClient`:

- `KmipClient` - we can leave it as is, since we can differentiate the interface and the class via imports
- `ProxyKmipClient` - designates that this client leverages the original proxy client underneath
- `PieKmipClient` - the first client representing the Pie API, seems odd to reference the package name in the client name but eh, whatever
- `BasicKmipClient` - designates that this client provides a simpler interface than the prior proxy client
- `MinimalKmipClient` - designates that this client only provides basic CRUD operations for KMIP managed objects

I think users would be scared to use a `HazardousKmipClient`, just my $0.02.

Thoughts? Preferences? It's just a name find/replace to me. I kind of like `ProxyKmipClient` but `BasicKmipClient` and `MinimalKmipClient` would also work fine. Also tagging @joel-coffman in case he wants to chime in.